app/serializers/component_serializer/status_component_serializer.rb

```diff
@@ -0,0 +1,29 @@
+module ComponentSerializer
+  class StatusComponentSerializer < ComponentSerializer::BaseComponentSerializer
+    # Initialise a status component.
+    #
+    # @param [String] type a string that represents the type for the status component.
+    # @param [Array<Hash>] display_data a hash containing display data.
+    # @param [String] selector a string to be the id attribute of the status component.
+    # @param [Array<Object>] components objects that are to be children of the status component.
+    def initialize(type: nil, display_data: nil, selector: nil, components: nil)
+      @type = type
+      @display_data = display_data
+      @selector = selector
+      @components = components
+    end
+
+    def name
+      "status__#{@type}"
+    end
+
+    def data
+      {}.tap do |hash|
+        hash[:link] = @link if @link
+        hash[:display] = { data: @display_data } if @display_data
```

Comment:
I see this line a lot across serialisers - is there anything we can do to dry it up that makes sense?